### PR TITLE
Feat/gen3users

### PR DESCRIPTION
**Pull request by paulineribeyre**
### New Features
- user.yaml validation code, Pypi setup, auto release notes

### CODE REVIEW

**Comment by paulineribeyre** [comment]
Tested in `commons-users`:
https://github.com/uc-cdis/commons-users/commit/60333e308e8a6f0a0875b470ff6cdae263aa0ae4
https://travis-ci.com/uc-cdis/commons-users/builds/112133891

**Comment by paulineribeyre** [comment]
Possible future improvements/additions:
- print all the errors over the whole file instead of stopping at the first error
- for now `anonymous_policies` and `all_users_policies` are not checked
- script to convert user.yaml to another format
- script to convert old user.yaml to new user.yaml / to generate smart policies/groups based on common user access

**Comment by PlanXCyborg** [comment]
The style in this PR agrees with [`black`](https://github.com/ambv/black). :heavy_check_mark:

This formatting comment was generated automatically by a script in [uc-cdis/wool](https://github.com/uc-cdis/wool).

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+- if [ $(python setup.py --version) == '0.0.0' ]; then travis_terminate 1; fi
+deploy:
+  provider: pypi
+  user: uc-ctds
+  skip_existing: true
+  skip_cleanup: true
+  on:
+    tags: true
+after_deploy:
+- "./release_notes.sh"
```
Comment (fantix) [inline .travis.yml]: Oh, I'm sorry we don't need this file anymore. It's now just:

```yaml
after_deploy:
- pip install gen3git
- gen3git release
```

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+- pip install pipenv
+- pipenv install --dev --deploy --python `which python`
+- pip install -e git+https://git@github.com/uc-cdis/wool.git#egg=wool
+script:
+- if [[ $PR_NUMBER != false ]]; then GITHUB_TOKEN=$GH_TOKEN wool; fi
+before_deploy:
+- sed -i.bak "s/=get_version()/='$TRAVIS_TAG'/g" setup.py
+- cat setup.py
+- if [ $(python setup.py --version) == '0.0.0' ]; then travis_terminate 1; fi
+deploy:
```
Comment (fantix) [inline .travis.yml:15]: The PyPI password secret here ;)

Comment (paulineribeyre) [reply .travis.yml:15]: oh, right :p

File: README.md [hunk 3]
```diff
@@ -1,3 +1,15 @@
...
+
+How to validate a `user.yaml` file:
+```
+from gen3users.validation import validate_user_yaml
+
+file_name = "my-gen3-commons/user.yaml"
+with open(file_name, "r") as f:
+    user_yaml = f.read()
+    validate_user_yaml(user_yaml, file_name)
+```
```
Comment (fantix) [inline README.md:15]: Can we make this a [CLI](https://python-packaging.readthedocs.io/en/latest/command-line-scripts.html#the-console-scripts-entry-point) please?


```bash
$ pip install gen3users
$ gen3users validate my-gen3-commons/user.yaml
```

File: setup.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    try:
+        tag = check_output(
+            ["git", "describe", "--tags", "--abbrev=0", "--match=[0-9]*"]
+        )
+        return tag.decode("utf-8").strip("\n")
+    except Exception as e:
+        raise RuntimeError(
+            "{}\nThe version number cannot be extracted from git tag in this source "
+            "distribution; please either download the source from PyPI, or check out "
+            "from GitHub and make sure that the git CLI is available.".format(e)
```
Comment (fantix) [inline setup.py]: maybe `raise RuntimeError(...) from e`?

**Review by fantix — APPROVED** [review]

**Pull request merged**